src/plugins/discover/public/application/helpers/get_switch_index_pattern_app_state.ts

```diff
@@ -36,23 +36,38 @@ import { IndexPattern } from '../../opensearch_dashboards_services';
  * Helper function to remove or adapt the currently selected columns/sort to be valid with the next
  * index pattern, returns a new state object
  */
+
+export interface PointInTime {
+  attributes: any;
+  id: string;
+}
 export function getSwitchIndexPatternAppState(
-  currentIndexPattern: IndexPattern,
-  nextIndexPattern: IndexPattern,
+  currentIndexPattern: IndexPattern | PointInTime,
+  nextIndexPattern: IndexPattern | PointInTime,
   currentColumns: string[],
   currentSort: SortPairArr[],
   modifyColumns: boolean = true
 ) {
-  const nextColumns = modifyColumns
-    ? currentColumns.filter(
-        (column) =>
+  if (nextIndexPattern.attributes) {
```

Comment:
This is a crude way to check whether it's a index pattern or point in time. Will have to do a better implementation for the same